File: lock/src/main/java/com/weiyi/lock/Person.java
package com.weiyi.lock;

import android.util.Log;

/**
 * @author TW
 * @date 2020/7/29 17:36
 * @description
 * @mail 2278671454@qq.com
 */
public class Person {
    private String name;
    private int age;

    public Person(String name, int age) {
        this.name = name;
        this.age = age;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }

}

File: roomlearn/src/main/java/com/weiyi/roomlearn/entity/Person.kt
package com.weiyi.roomlearn.entity

import androidx.room.ColumnInfo
import androidx.room.Embedded
import androidx.room.Entity
import androidx.room.PrimaryKey

/**
 * @author TW
 * @date 2020/6/3 11:50
 * @description
 * @mail 2278671454@qq.com
 */
//@Entity(
////        indices = [Index(value = ["username", "idCard"], name = "identity", unique = true)],
////       primaryKeys = ["id", "idCard"],
//        tableName = PERSON_TABLE
//)
@Entity
data class Person(@ColumnInfo(name = "username")
                  var name: String = "") {
    @PrimaryKey(autoGenerate = true)
    var id: Long = 0

    @ColumnInfo
    var idCard = ""

    @ColumnInfo
    var location: String = ""

    @ColumnInfo
    var age: Int = 0

    @ColumnInfo
    //0男，1女
    var sex: Int = 0

    //身份证照片
//    @Ignore
//    var picture: Bitmap? = null

    @Embedded(prefix = "first")
    var message1: PersonMessage? = null


    @Embedded(prefix = "second")
    var message2: PersonMessage? = null

    private fun getSex(sex: Int): String {
        return if (sex == 0) {
            "男"
        } else {
            "女"
        }
    }

    override fun toString(): String {
        return "name:$name;age:$age;sex:${getSex(sex)};location:$location;idCard:$idCard;marriage:${message1?.marriage}}\n"
    }
}